src/NewTwitchApi/Resources/EventSubApi.php

```diff
@@ -0,0 +1,262 @@
+<?php
+
+declare(strict_types=1);
+
+namespace NewTwitchApi\Resources;
+
+use GuzzleHttp\Client;
+use GuzzleHttp\Exception\GuzzleException;
+use NewTwitchApi\HelixGuzzleClient;
+
+class EventSubApi
+{
+    private $clientId;
+    private $secret;
+    private $guzzleClient;
+
+    public function __construct(string $clientId, string $secret, Client $guzzleClient = null)
+    {
+        $this->clientId = $clientId;
+        $this->secret = $secret;
+        $this->guzzleClient = $guzzleClient ?? new HelixGuzzleClient($clientId);
+    }
+
+    /**
+     * @throws GuzzleException
+     * @link https://dev.twitch.tv/docs/eventsub
+     */
+    private function subscribe(string $type, string $version, array $condition, string $callback, string $bearer): void
+    {
+        $headers = [
+            'Client-ID' => $this->clientId,
+        ];
+
+        $headers['Authorization'] = sprintf('Bearer %s', $bearer);
+
+        $body = [
+            'type' => $type,
+            'version' => $version,
+            'condition' => $condition,
+            'transport' => [
+                'method' => 'webhook',
+                'callback' => $callback,
+                'secret' => $this->secret,
+            ],
+        ];
+
+        $this->guzzleClient->post('eventsub/subscriptions', [
```

Comment:
We should continue using the helper functions in the Abstract Resource as we do for the rest of our classes. It is likely we need to change how the Abstract Resource handles Bodies as we had issues with tests previously, so feel free to recommend some changes as you get your tests to process.